celery/concurrency/solo.py

```diff
@@ -1,29 +1,47 @@
 """Single-threaded execution pool."""
+from __future__ import annotations
+
 import os
+from typing import TYPE_CHECKING, Any
 
 from celery import signals
 
 from .base import BasePool, apply_target
 
-__all__ = ('TaskPool',)
+__all__ = ("TaskPool",)
+
+if TYPE_CHECKING:
+    from typing import TypedDict
+
+    PoolInfo = TypedDict(
+        "PoolInfo",
+        {
+            "max-concurrency": int,
+            "processes": list[int],
+            "max-tasks-per-child": None,
+            "put-guarded-by-semaphore": bool,
+            "timeouts": tuple,
+        },
+    )
 
 
 class TaskPool(BasePool):
     """Solo task pool (blocking, inline, fast)."""
+    limit: int
 
     body_can_be_buffer = True
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
         super().__init__(*args, **kwargs)
         self.on_apply = apply_target
         self.limit = 1
         signals.worker_process_init.send(sender=None)
 
-    def _get_info(self):
+    def _get_info(self) -> PoolInfo:
         return {
-            'max-concurrency': 1,
-            'processes': [os.getpid()],
-            'max-tasks-per-child': None,
-            'put-guarded-by-semaphore': True,
-            'timeouts': (),
+            "max-concurrency": 1,
+            "processes": [os.getpid()],
+            "max-tasks-per-child": None,
+            "put-guarded-by-semaphore": True,
+            "timeouts": (),
```

Comment:
Side effect from my IDE. If it bothers, accept the following suggestion:
```suggestion
            'max-concurrency': 1,
            'processes': [os.getpid()],
            'max-tasks-per-child': None,
            'put-guarded-by-semaphore': True,
            'timeouts': (),
```